Azure/cloud-init.txt

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Description: This script will install all dependencies
+# required by clear containers and metrics/tests.
+#
+# OS: ubuntu 16.04
+
+# Add repository key
+wget -nv https://download.opensuse.org/repositories/home:clearcontainers:clear-containers-3/xUbuntu_16.04/Release.key -O Release.key
+apt-key add - < Release.key
+apt-get update
+
+# Add repositories and install runtime
+echo "Installing Clear Containers"
+sh -c "echo 'deb http://download.opensuse.org/repositories/home:/clearcontainers:/clear-containers-3/xUbuntu_16.04/ /' > /etc/apt/sources.list.d/cc-runtime.list"
+apt-get update
+apt-get install -y cc-runtime
+
+
+# Install docker
+echo "Installing docker"
+apt-get install -y \
+	apt-transport-https \
+	ca-certificates \
+	curl \
+	software-properties-common
+
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+apt-key fingerprint 0EBFCD88
+
+add-apt-repository \
+	"deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+	$(lsb_release -cs) \
+	stable"
+
+apt-get update
+apt-get install -y docker-ce
+
+# Install tests dependencies
+apt-get install -y \
+	bc \
+	smem \
+	apache2-utils \
+	iperf
+
+# Install thin tools for devicemapper configuration
+apt-get install -y thin-provisioning-tools
+
+# Configure the clear containers runtime using devicemapper as
+# storage driver
+mkdir -p /etc/systemd/system/docker.service.d/
+cat << EOF | sudo tee /etc/systemd/system/docker.service.d/clr-containers.conf
+[Service]
+ExecStart=
+ExecStart=/usr/bin/dockerd -D --add-runtime cc-runtime=/usr/bin/cc-runtime --default-runtime=cc-runtime
+EOF
+
+# Add devicemapper configuration
+mkdir -p /etc/docker
+
+mkdir -p /etc/lvm/profile
+cat << EOF | sudo tee /etc/docker/daemon.json
+{
+"storage-driver": "devicemapper",
+"storage-opts": [
+        "dm.directlvm_device=/dev/sdc",
+        "dm.thinp_percent=95",
+        "dm.thinp_metapercent=1",
+        "dm.thinp_autoextend_threshold=80",
+        "dm.thinp_autoextend_percent=20",
+        "dm.directlvm_device_force=true"
+]
+}
```

Comment:
you could also move the cc-runtime adding to here as well, just so it's all in one spot but up to you:

```
{
	"default-runtime": "cc-runtime",
	"runtimes": {
		"runc": {
			"path": "runc"
		},
		"custom": {
			"path": "/usr/bin/cc-runtime",
		}
	}
}
```

Reply:
well, both ways are valid/correct, I think would be better to have this conf in different places, thanks for advise :)

Reply:
@jessfraz this had failed for me when I tried it initially.

This is the preferred option as this will ensure that we do not overwrite the default setup for the remaining parameters as installed. Will try this again.

Reply:
oh interesting, I don't care either way just thought I'd mention it :)

Reply:
@jessfraz I meant the method you mentioned is the preferred option. Not what we have currently.

```
{
	"default-runtime": "cc-runtime",
	"runtimes": {
		"runc": {
			"path": "runc"
		},
		"custom": {
			"path": "/usr/bin/cc-runtime",
		}
	}
}
```

Reply:
The working daemon.json file needs to be as follows (i.e runc needs to be excluded and will be picked up by default
```
{
"storage-driver": "devicemapper",
"storage-opts": [
        "dm.directlvm_device=/dev/sdc",
        "dm.thinp_percent=95",
        "dm.thinp_metapercent=1",
        "dm.thinp_autoextend_threshold=80",
        "dm.thinp_autoextend_percent=20",
        "dm.directlvm_device_force=true"
],
       "default-runtime": "cc-runtime",
        "runtimes": {
             "cc-runtime": {
                      "path": "/usr/bin/cc-runtime"
             }
        }
}
```

Adding runc results in an error 
```
/etc/docker/daemon.json: configuration validation from file failed (runtime name 'runc' is reserved)
```

Reply:
https://github.com/docker/for-linux/issues/144